### PR TITLE
Feature/allow variable for legends

### PR DESCRIPTION
**Pull request by elenatorro**
Fix https://github.com/CartoDB/cartoframes/issues/939
---

This PR allows using a `variable` from the viz to get the legend data

### CODE REVIEW

File: packages/bridge/src/vl/legends/index.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ function _getColorValue(viz, propName) {
     return rgbaToString(_getLegendData(prop));
   }
 
-  if (propName === 'color' || propName === 'strokeColor') {
+  if (prop.type === 'color') {
```
Comment (jesusbotella) [inline packages/bridge/src/vl/legends/index.ts]: I think I'd need more context on this change :)

File: packages/bridge/src/vl/legends/index.ts [hunk 2]
```diff
@@ -39,8 +39,10 @@ function _getNumberValue(viz, propName, defaultValue?) {
   return defaultValue;
 }
 
-function _getSymbolValue(viz) {
-  const prop = viz.symbol;
+function _getSymbolValue(viz, value) {
```
[the remark this answers is not in the document]
Comment (elenatorro) [reply packages/bridge/src/vl/legends/index.ts]: You're right! let's use `propertyName`

File: packages/bridge/src/vl/legends/index.ts (hunk 1 above)
Comment (elenatorro) [reply packages/bridge/src/vl/legends/index.ts]: Now that we're allowing to use a VL variable, we should check if the property from the viz (the style property, or the variable) has a 'color' type. For example, having:

```
@my_var: blue
color: blue
` ``

Both `@my_var` and `color` type will be color.

File: packages/bridge/src/vl/legends/index.ts [hunk 3]
```diff
@@ -171,8 +173,14 @@ export default class Legends {
       const baseStyle = _styleFromLayer(parsedLayer);
 
       const vizProp = parsedLayer.layer.viz[prop];
+
       const config = options.config;
-      const legendData = parsedLayer.layer.viz[prop].getLegendData(config).data;
+      const dataProp = options.variable || prop;
```
[the remark this answers is not in the document]
Comment (elenatorro) [reply packages/bridge/src/vl/legends/index.ts]: yes, totally right 👍 I'll fix that

**Comment by elenatorro** [comment]
Moving this PR to https://github.com/CartoDB/airship/pull/649

**Pull request closed without merging**